src/gluonts/model/deepstate/_estimator.py

```diff
@@ -153,7 +153,7 @@ def __init__(
         self,
         freq: str,
         prediction_length: int,
-        cardinality: List[int],
+        cardinality: Optional[List[int]] = None,
```

Comment:
Now you cannot construct the estimator with default values since `cardinality` and `use_feat_static_cat` (which is `True` by default) are in conflict. This was why we made this argument non-optional. 

I think we agreed that when this estimator is used with `use_feat_static_cat=False`, then the user explicitly sets `cardinality = [1]` (which is the right thing to do) and is an additional check so that it is not misused (this case should not happen most of the times). We should update the docstring for `cardinality` argument to reflect this.

Reply:
Agreed, let's update the docstring instead then

Reply:
@rshyamsundar on a second thought: I think the way I fixed it in this PR just makes things run smoothly in case you set `use_feat_static_cat=False`. With default values (`use_feat_static_cat=True`) you would get a validation error prior to this PR, and an assertion error after this PR.

Reply:
> if no cardinality is provided then you would get a validation error prior to this PR,

`cardinality` was non-optional argument prior to this PR, so it has to be given by default like `freq` or `pred_length` argument. Prior to this PR, this estimator can be instantiated by providing only the non-optional arguments but now if one instantiates that way they get an assertion error.